### ISSUE TITLE
[BUG][CRASH] Segfault while using editor.

### ISSUE THREAD

**Issue by Icantjuddle** [mawww/kakoune#4830]
### Version of Kakoune

Kakoune v2021.11.08-135-geae8ea8a

### Reproducer

I don't know how this came about, if it happens again I will update. 

### Outcome

Callstack that I put through `llvm-cxxfilt`.
```
 1│Received SIGSEGV, exiting.
 2│Pid: 2797609
 3│Callstack:
 4│kak(Kakoune::Backtrace::Backtrace()+0x18) [0x4d96b8]
 5│kak(Kakoune::signal_handler(int)+0x6d) [0x5dc20d]
 6│/lib/x86_64-linux-gnu/libc.so.6(+0x430c0) [0x7f0b21d230c0]
 7│kak(Kakoune::Buffer::advance(Kakoune::BufferCoord, Kakoune::ByteCount) const+0x6d) [0x4dea4d]
 8│kak(Kakoune::InsertCompleter::select(int, bool, std::__1::vector<Kakoune::Key, Kakoune::Allocator<Kakoune::Key, (Kakoune::MemoryDomain)0> >&)+0x3da) [0x5b033a]
 9│kak(Kakoune::InputModes::Insert::on_key(Kakoune::Key)+0x890) [0x59fb50]
10│kak(Kakoune::InputHandler::handle_key(Kakoune::Key)+0x43b) [0x597eeb]
11│kak(Kakoune::Client::process_pending_inputs()+0x2bc) [0x4ee45c]
12│kak(Kakoune::ClientManager::process_pending_inputs()+0x73) [0x4f7973]
13│kak(Kakoune::run_server(Kakoune::StringView, Kakoune::StringView, Kakoune::StringView, Kakoune::Optional<Kakoune::BufferCoord>, Kakoune::ServerFlags, Kakoune::UIType, Kakoune::DebugFlags, Kakoune::ArrayView<Kakoune::St
14│kak(main+0x17a0) [0x5ddbf0]
15│/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7f0b21d040b3]
16│kak(_start+0x2e) [0x4d7a1e]
```

### Additional information

Running on Ubuntu 20.04

Built with the following script:
```
#!/bin/bash

export PREF="$HOME/.local/apps/clang/clang+llvm-12.0.0-x86_64-linux-gnu-ubuntu-20.04/"
export CXX="$PREF/bin/clang++"
export CPPFLAGS+="-nostdinc++ -nostdlib++ -isystem $PREF/include/c++/v1 -march=native"
export LDFLAGS+="-L$PREF/lib -Wl,-rpath,$PREF/lib -lc++"
make -j
```

**Comment by mawww** [comment]
I believe this has been fixed by 293d46837d1d28e40f3d0ddf21c653aa616bbf40

**Comment by Icantjuddle** [comment]
Thank you!
